### PR TITLE
(WIP) - Promotion setup and an environment file

### PR DESCRIPTION
**Pull request by Wesley201**
# Peer Review Information:
Changing the promotion file and publish, publish trigger on any branch.
Promotion requires the release tag for the promotion of the package from the internal server
Updated the environments file so all yml files are using it  

## Editor / Package versioning:
**Editor Version Target (i.e. 19.3, 20.1)**: 2019.3
2019.3

## Dev Testing:
**Tests Added**: NO
<br>
**Package Tests (Pass/Fail)**: 
[X] - Make sure automation passes 
<br>
**Core Scenario Tested**: CI
<br>
**At Risk Areas**: Promotion and publish
<br>
**Notes + Expectations**:

### CODE REVIEW

File: .yamato/promotion.yml [hunk 1]
```diff
@@ -1,79 +1,71 @@
...
+promotion_test_trigger:
+  name: Promotion Tests Trigger
+  dependencies:
+{% for editor in publish_editors %}
+{% for platform in publish_platforms %}
+{% for suite in suites %}
+{% for project in projects %}
+    - .yamato/promote.yml#promotion_test_{{platform.name}}_{{editor.version}}
+    - .yamato/upm-ci-full.yml#pkg_test_{{variant.name}}_{{platform.name}}_{{editor.version}}
+    - .yamato/upm-ci-testprojects#{{project.name}}_windows_{{suite.name}}_{{editor.version}}.yml#
```
Comment (JonathanHUnity) [inline .yamato/promotion.yml:38]: the .yml here looks off

File: .yamato/publish.yml [hunk 2]
```diff
@@ -9,12 +9,12 @@ publish:
...
     flavor: b1.large
   commands:
-    - npm install upm-ci-utils@stable -g --registry https://artifactory.prd.cds.internal.unity3d.com/artifactory/api/npm/upm-npm
+    - npm install upm-ci-utils@stable -g --registry {{ upmci_registry }}
     - upm-ci package publish --package-path com.unity.perception
   triggers:
     tags:
       only:
-        - /^(r|R)(c|C)-\d+\.\d+\.\d+(-preview(\.\d+)?)?$/
+        - "/^(r|R)(c|C)-\d+\.\d+\.\d+(-preview(\.\d+)?)?$/"
```
Comment (JonathanHUnity) [inline .yamato/publish.yml:17]: I don't think you should need the quotes

Comment (Wesley201) [reply .yamato/publish.yml]: fixed

File: .yamato/promotion.yml (hunk 1 above)
Comment (Wesley201) [reply .yamato/promotion.yml]: fixed

File: .yamato/publish.yml [hunk 3]
```diff
@@ -9,12 +9,12 @@ publish:
...
     flavor: b1.large
   commands:
-    - npm install upm-ci-utils@stable -g --registry https://artifactory.prd.cds.internal.unity3d.com/artifactory/api/npm/upm-npm
+    - npm install upm-ci-utils@stable -g --registry {{ upmci_registry }}
     - upm-ci package publish --package-path com.unity.perception
   triggers:
     tags:
       only:
-        - /^(r|R)(c|C)-\d+\.\d+\.\d+(-preview(\.\d+)?)?$/
+        - "/^(r|R)(c|C)-+(-preview(+)?)?$/"
```
Comment (JonathanHUnity) [inline .yamato/publish.yml:17]: This still doesn't look like the proper regex. The original one looks right to me.
```suggestion
        - /^(r|R)(c|C)-\d+\.\d+\.\d+(-preview(\.\d+)?)?$/
```

**Pull request merged**